addons/udes_sale_stock/models/sale_order_line.py

```diff
@@ -9,19 +11,31 @@ class SaleOrderLine(models.Model):
     is_cancelled = fields.Boolean(string='Cancelled', readonly=True, store=True,
                                   default=False, index=True,
                                   compute='_compute_is_cancelled')
+    cancel_date = fields.Datetime(string='Cancel Date',
+                                  help='Time of cancellation',
+                                  readonly=True, index=True)
 
     @api.depends('move_ids.state')
     def _compute_is_cancelled(self):
         location_customers = self.env.ref('stock.stock_location_customers')
+        now_date = datetime.now()
+
+        def not_cancelled_filter(m):
```

Comment:
just as a question, do we like to do this kind of defining rather than using a lambda? I only see lambdas in the code so I dno if there is a reason for it or a preference or neither as the reason why.